src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -43,11 +44,18 @@
 #define stringify(x)  #x
 #define expand_and_stringify(x) stringify(x)
 
-// Path from site-packages to packages that contain NuPIC Python regions
-static std::vector<const char *> packages { "nupic.regions", "nupic.regions.extra" };
-
 namespace nupic
 {
+  // Path from site-packages to packages that contain NuPIC Python regions
+  static std::vector<const char *> packages { "nupic.regions", "nupic.regions.extra" };
+
+  // Mappings for c++ regions
+  static std::map<const std::string, GenericRegisteredRegionImpl*> cpp_packages;
```

Comment:
I just added code to delete the GenericRegisteredRegionImpl when the RegionImplFactory is no longer necessary